### PR TITLE
autopilot+config: support opening private channels with autopilot agent

### PR DESCRIPTION
**Pull request by wpaulino**

### CODE REVIEW

File: autopilot/agent_test.go [hunk 1]
```diff
@@ -704,6 +708,132 @@ func TestAgentImmediateAttach(t *testing.T) {
...
+		// establish more channels, and give it a budget of 5 BTC to do
+		// so.
+		resp := moreChansResp{
+			needMore: true,
+			numMore:  numChans,
+			amt:      5 * btcutil.SatoshiPerBitcoin,
+		}
+		select {
+		case heuristic.moreChansResps <- resp:
+			wg.Done()
```
Comment (halseth) [inline autopilot/agent_test.go]: just add a defer in the beginning.

File: config.go [hunk 2]
```diff
@@ -146,6 +146,7 @@ type autoPilotConfig struct {
 	Allocation     float64 `long:"allocation" description:"The percentage of total funds that should be committed to automatic channel establishment"`
 	MinChannelSize int64   `long:"minchansize" description:"The smallest channel that the autopilot agent should create"`
 	MaxChannelSize int64   `long:"maxchansize" description:"The largest channel that the autopilot agent should create"`
+	Private        bool    `long:"private" description:"Whether the channels created by the autopilot agent should be announced to the network or not"`
```
Comment (halseth) [inline config.go]: Description states the negative of the actual flag. Could say ""Whether the channels created by the autopilot agent should be private or not. Private channels won't be announced to the network."

File: autopilot/agent_test.go (hunk 1 above)
Comment (wpaulino) [reply autopilot/agent_test.go]: Fixed.

File: config.go (hunk 2 above)
Comment (wpaulino) [reply config.go]: Fixed.

**Comment by cfromknecht** [comment]
needs rebase

**Comment by wpaulino** [comment]
Rebased.

**Pull request merged**